components/forms/newsletter/component.jsx

```diff
@@ -30,32 +55,35 @@ class NewsletterForm extends PureComponent {
     initialValues: PropTypes.object,
   };
 
-  saveNewsletterSubscription = (values) => {
+  saveNewsletterSubscription = async (values) => {
+    const ipAddress = await fetch('https://api.ipify.org/?format=json')
```

Comment:
noob question, what is this url for?

Reply:
Not a noob question, this is a weird one. 

Task requirements list that we should send the _client_'s IP address as as a field when submitting the form.   
This is odd, but in order to do so (within the timeframe), when submitting the form, I'm pinging a third party service to get the ip address of the user (ran in the browser, so we are 100% sure it'll be the user's IP address). 

I went into specifics in the [JIRA task](https://gfw.atlassian.net/browse/FLAG-1014?focusedCommentId=24534) as to why this particular one, but it's basically because it's a free service, not rate limited, and the project is open source which fits. 

I do think a different solution should be procured in the future; and I'm not entirely sure sending the ip address is needed at all in the first place; Ortto should have this information already but this is a stepping stone.